### PR TITLE
cli: allow to provide single node config file

### LINKED ISSUE #2978: feature request: --config-file option to start node
The "feature" was discussed here https://github.com/nspcc-dev/neo-go/issues/2874#issuecomment-1381558863 and the time has come that I'm actually missing this (I'm planning to wrap `neo-go` so I can drop the need for `neo-express` to run examples with `neo-mamba`).

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Close #2978.

### CODE REVIEW

File: cli/options/options.go [hunk 1]
```diff
@@ -35,9 +35,9 @@ const RPCEndpointFlag = "rpc-endpoint"
 // Network is a set of flags for choosing the network to operate on
 // (privnet/mainnet/testnet).
 var Network = []cli.Flag{
-	cli.BoolFlag{Name: "privnet, p", Usage: "use private network configuration"},
-	cli.BoolFlag{Name: "mainnet, m", Usage: "use mainnet network configuration"},
-	cli.BoolFlag{Name: "testnet, t", Usage: "use testnet network configuration"},
+	cli.BoolFlag{Name: "privnet, p", Usage: "use private network configuration (if --config-path option is specified)"},
+	cli.BoolFlag{Name: "mainnet, m", Usage: "use mainnet network configuration (if --config-path option is specified)"},
+	cli.BoolFlag{Name: "testnet, t", Usage: "use testnet network configuration (if --config-path option is specified)"},
```
Comment (roman-khimov) [inline cli/options/options.go:40]: That's not exactly the case. It's rather `if --config-file is not specified`. I'd expect `neo-go node -m` to still work the same way as before.

**Review by roman-khimov — CHANGES_REQUESTED** [review]

File: cli/options/options.go [hunk 2]
```diff
@@ -63,7 +63,14 @@ var Historic = cli.StringFlag{
 // Config is a flag for commands that use node configuration.
 var Config = cli.StringFlag{
 	Name:  "config-path",
-	Usage: "path to directory with configuration files",
+	Usage: "path to directory with configuration files (may be overridden by --config-file option)",
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply cli/options/options.go]: Fixed.

File: cli/options/options.go [hunk 3]
```diff
@@ -63,7 +63,14 @@ var Historic = cli.StringFlag{
...
+}
+
+// ConfigFile is a flag for commands that use node configuration and provide
+// path to the specific config file instead of config path.
+var ConfigFile = cli.StringFlag{
+	Name:  "config-file",
+	Usage: "path to a single node configuration file (overrides --config-path option)",
```
[the remark this answers is not in the document]
Comment (roman-khimov) [reply cli/options/options.go]: It's just a configuration file. It inevitably has some network (magic) inside of it.

File: cli/options/options.go [hunk 4]
```diff
@@ -152,7 +159,11 @@ func GetRPCWithInvoker(gctx context.Context, ctx *cli.Context, signers []transac
 // GetConfigFromContext looks at the path and the mode flags in the given config and
 // returns an appropriate config.
 func GetConfigFromContext(ctx *cli.Context) (config.Config, error) {
-	configPath := "./config"
+	configFile := ctx.String("config-file")
+	if len(configFile) != 0 {
+		return config.LoadFile(configFile)
+	}
+	var configPath = "./config"
```
Comment (roman-khimov) [inline cli/options/options.go:166]: I do love `var` definitions, but I also love consistency. Here you change from `:=` to `var` while introducing another `:=`. It'd be nice to have some similarity.

File: cli/options/options.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply cli/options/options.go]: Finally, it's just a `node configuration file`, fixed everywhere.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**